src/System.Windows.Forms/src/System/Windows/Forms/ContainerControl.cs

```diff
@@ -1326,6 +1353,83 @@ private void ScrollActiveControlIntoView()
             }
         }
 
+        protected override void RescaleConstantsForDpi(int deviceDpiOld, int deviceDpiNew)
+        {
+            if (deviceDpiNew == deviceDpiOld)
+            {
+                return;
+            }
+
+            base.RescaleConstantsForDpi(deviceDpiOld, deviceDpiNew);
+
+            // Check if font is inherited from parent and is not being scaled by Parent (e.g. Winforms designer
+            // in Visual Studio). In this case we need to scale Control explicitly with respect to new scaled Font.
+            if (TryGetExplicitlySetFont(out _))
+            {
+                return;
+            }
+
+            using (new LayoutTransaction(ParentInternal, this, PropertyNames.Font))
+            {
+                OnFontChanged(EventArgs.Empty);
+            }
+        }
+
+        internal void ScaleContainer(int deviceDpiNew, int deviceDpiOld, Rectangle suggestedRectangle)
+        {
+            CommonProperties.xClearAllPreferredSizeCaches(this);
+            SuspendAllLayout(this);
+            try
+            {
+                // If this container is a top-level window, we would receive WM_DPICHANGED message that
+                // has SuggestedRectangle for the control. We are forced to use this in such cases to
+                // make the control placed in right location with respect to the new monitor that triggered
+                // WM_DPICHANGED event. Failing to apply SuggestedRectangle will result in a circular WM_DPICHANGED
+                // events on the control.
+
+                // Note: SuggestedRectangle supplied  by WM_DPICHANGED event is Dpi (not Font) scaled. if top-level window is
+                // Font scaled, we might see deviations in the expected bounds and may result in adding Scrollbars (horizantal/vertical)
+                User32.SetWindowPos(
+                    new HandleRef(this, HandleInternal),
+                    User32.HWND_TOP,
+                    suggestedRectangle.X,
+                    suggestedRectangle.Y,
+                    suggestedRectangle.Width,
+                    suggestedRectangle.Height,
+                    User32.SWP.NOZORDER | User32.SWP.NOACTIVATE);
```

Comment:
I think this can be simplified to:
```suggestion
                User32.SetWindowPos(
                    this,
                    User32.HWND_TOP,
                    suggestedRectangle.X,
                    suggestedRectangle.Y,
                    suggestedRectangle.Width,
                    suggestedRectangle.Height,
                    User32.SWP.NOZORDER | User32.SWP.NOACTIVATE);
```

Reply:
This was an existing call moved around and consistence with other use cases . HandleRef supposed to keep track of ref count before disposing. I will have to  understand this more before changing. May look into it in the next PR.

Reply:
`new HandleRef(this, HandleInternal)` is an overhead, which we deal with via `IHandle` interface.
Also `this, HandleInternal` is the same as `this.HandleInternal` which is the same as `this.Handle`, which is handled by `IHandle` :)